samples/sensor/accel_trig/sample.yaml

```diff
@@ -35,3 +35,29 @@ tests:
       - SNIPPET=rtt-tracing;rtt-console
     platform_allow:
       - apard32690/max32690/m4
+  sample.sensor.accel_trig.shield-tap:
+    harness: console
+    tags: sensors
+    harness_config:
+      type: one_line
+      regex:
+        - "^\\s*[0-9A-Za-z_,+-.]*@[0-9A-Fa-f]* \\[m\/s\\^2\\]:    \
+          \\(\\s*-?[0-9\\.]*,\\s*-?[0-9\\.]*,\\s*-?[0-9\\.]*\\)$"
+    extra_args:
+      - SHIELD=x_nucleo_iks01a3
+      - EXTRA_DTC_OVERLAY_FILE=x_nucleo_iks01a3.overlay
+      - CONFIG_SAMPLE_TAP_DETECTION=y
+    platform_allow:
+      - nrf52dk/nrf52832
+  sample.sensor.accel_trig.lis2dw12-tap:
+    harness: console
+    tags: sensors
+    harness_config:
+      type: one_line
+      regex:
+        - "^\\s*[0-9A-Za-z_,+-.]*@[0-9A-Fa-f]* \\[m\/s\\^2\\]:    \
+          \\(\\s*-?[0-9\\.]*,\\s*-?[0-9\\.]*,\\s*-?[0-9\\.]*\\)$"
+    extra_args:
+      - CONFIG_SAMPLE_TAP_DETECTION=y
```

Comment:
Any why is `CONFIG_SAMPLE_TAP_DETECTION=y` only expected to work on `stm32f3_disco`?

Reply:
`CONFIG_SAMPLE_TAP_DETECTION` only works on platforms with an accelerometer that supports a `SENSOR_TRIG_DOUBLE_TAP` trigger.

Reply:
Should the filtering be based on the devicetree compatible then, not a single platform?

Reply:
I think there is no way no detect the compatibility from the device tree.
The only way is by looking in driver sources.
```
nxp/fxos8700/fxos8700_trigger.c
st/iis2dlpc/iis2dlpc_trigger.c
tdk/icm42605/icm42605.c
```

Reply:
I am suggesting filtering based on the compatible of the driver, for example:
`    filter: dt_compat_enabled("st,stm32-aes") or dt_compat_enabled("st,stm32-cryp")`
You already have a list of drivers this sample should work with, so it is not limited to just `stm32f3_disco`

Reply:
Ok I have pushed a new version with your suggestion